pkg/apis/hive/v1/dnsendpoint_types.go

```diff
@@ -1,82 +0,0 @@
-package v1
-
-import (
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-)
-
-const (
-	// FinalizerDNSEndpoint is used on DNSEndpoints to ensure we successfully deprovision
-	// the cloud objects before cleaning up the API object.
-	FinalizerDNSEndpoint string = "hive.openshift.io/dnsendpoint"
-)
-
-// DNSEndpointSpec defines the desired state of DNSEndpoint
-type DNSEndpointSpec struct {
-	// Endpoints is the list of DNS records to create/update
-	// +optional
-	Endpoints []*Endpoint `json:"endpoints,omitempty"`
-}
-
-// DNSEndpointStatus defines the observed state of DNSEndpoint
-type DNSEndpointStatus struct {
-	// ObservedGeneration is the generation observed by the external-dns controller.
-	// +optional
-	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
-}
-
-// TTL is the time to live of DNS records
-type TTL int64
-
-// Targets is the set of values associated with a DNS record
-type Targets []string
-
-// Labels is a set of labels associated with a DNS record
-type Labels map[string]string
-
-// ProviderSpecific contains cloud provider specific configuration for a DNS record
-type ProviderSpecific map[string]string
-
-// Endpoint represents a single DNS record
-type Endpoint struct {
-	// The hostname of the DNS record
-	DNSName string `json:"dnsName,omitempty"`
-	// The targets the DNS record points to
-	Targets Targets `json:"targets,omitempty"`
-	// RecordType type of record, e.g. CNAME, A, SRV, TXT etc
-	RecordType string `json:"recordType,omitempty"`
-	// TTL for the record
-	RecordTTL TTL `json:"recordTTL,omitempty"`
-	// Labels stores labels defined for the Endpoint
-	// +optional
-	Labels Labels `json:"labels,omitempty"`
-	// ProviderSpecific stores provider specific config
-	// +optional
-	ProviderSpecific ProviderSpecific `json:"providerSpecific,omitempty"`
```

Comment:
This data doesn't seem to have been moved to DNSZone, I assume it just wasn't necessary for us to track?

Reply:
Correct. And that is my main motivation for removing the DNSEndpoint CRD. It is filled with fields that Hive does not care about--or that will always be the same value for Hive. After removing all of the unnecessary fields from the DNSEndpoint CRD, we would be left with just the `DNSName` and `Targets` fields, which are both already present in the the DNSZone CRD.